apps/settings/js/panels/addon_details/panel.js

```diff
@@ -29,10 +33,14 @@ define(function(require) {
         this._elements.toggle.onchange = this._onToggleChange.bind(this);
         // Hook up the delete button
         this._elements.deleteButton.onclick = this._onDelete.bind(this);
+        // Hook up the rename button
+        this._elements.renameButton.onclick = this._onRename.bind(this);
       },
 
       onBeforeShow: function(panel, options) {
         return Promise.resolve((() => {
+          // Renaming should only be available from activity.
+          this._details.isActivity = !options.addon;
```

Comment:
I would suggest to pass the value to the `render` function rather than attaching it here.